lib/deimos/active_record_consume/batch_consumption.rb

```diff
@@ -163,6 +212,40 @@ def compact_messages(batch)
 
         batch.reverse.uniq(&:key).reverse!
       end
+
+      # Turns Kafka payload into ActiveRecord Objects by mapping relevant fields
+      # Override this method to build object and associations with message payload
+      # @param messages [Array<Deimos::Message>] the array of deimos messages in batch mode
+      # @return [Array<ActiveRecord>] Array of ActiveRecord objects
+      def build_records(messages)
+        # This method must be overriden when association_list is configured
+        raise NotImplementedError unless @association_list.blank?
```

Comment:
Probably should have a better error message here. Also, I learned that `NotImplementedError` is actually the wrong thing to use here. In other repos I had a `MissingImplementationError` - if we don't have one, we probably should create one and move any other `NotImplementedError` cases to use it.

Reply:
I made MissingImplementationError in deimos/lib/exceptions.rb. 
There are very many NotImplementedError instances in Deimos. To avoid too many files changes, after this review is complete, I can add it as a separate commit in this PR.